tree.go

```diff
@@ -0,0 +1,125 @@
+
+		containerConfig, _, err := metadata.ReadContainerCheckpointConfigDump(task.outputDir)
+		if err != nil {
+			return err
+		}
+		specDump, _, err = metadata.ReadContainerCheckpointSpecDump(task.outputDir)
+		if err != nil {
+			return err
+		}
+
+		ci, err = getContainerInfo(task.outputDir, specDump, containerConfig)
+		if err != nil {
+			return err
+		}
+
+		archiveSizes, err := getArchiveSizes(task.checkpointFilePath)
+		if err != nil {
+			return err
+		}
+
+		var tree treeprint.Tree
+
+		if ci.Name == "" {
+			ci.Name = "Container"
+		}
+
+		tree = treeprint.NewWithRoot(ci.Name)
+
+		tree.AddBranch(fmt.Sprintf("Image: %s", containerConfig.RootfsImageName))
+		tree.AddBranch(fmt.Sprintf("ID: %s", containerConfig.ID))
+		tree.AddBranch(fmt.Sprintf("Runtime: %s", containerConfig.OCIRuntime))
+		tree.AddBranch(fmt.Sprintf("Created: %s", ci.Created))
+		tree.AddBranch(fmt.Sprintf("Engine: %s", ci.Engine))
+
+		if ci.IP != "" {
+			tree.AddBranch(fmt.Sprintf("IP: %s", ci.IP))
+		}
+		if ci.MAC != "" {
+			tree.AddBranch(fmt.Sprintf("MAC: %s", ci.MAC))
+		}
+
+		tree.AddBranch(fmt.Sprintf("CHKPT Size: %s", metadata.ByteToString(archiveSizes.checkpointSize)))
+
+		// Display root fs diff size if available
+		if archiveSizes.rootFsDiffTarSize != 0 {
+			tree.AddBranch(fmt.Sprintf("Root Fs Diff Size: %s", metadata.ByteToString(archiveSizes.rootFsDiffTarSize)))
+		}
+
+		if mounts {
+			mountsTree := tree.AddBranch("Mounts")
+			for _, data := range specDump.Mounts {
+				mountTree := mountsTree.AddBranch(fmt.Sprintf("Destination: %s", data.Destination))
+				mountTree.AddBranch(fmt.Sprintf("Type: %s", data.Type))
+				mountTree.AddBranch(fmt.Sprintf("Source: %s", func() string {
+					if fullPaths {
+						return data.Source
+					}
+					return shortenPath(data.Source)
+				}()))
+			}
+		}
+
+		if stats {
+			// Get dump statistics with crit
+			dumpStats, err := crit.GetDumpStats(tasks[0].outputDir)
+			if err != nil {
+				return fmt.Errorf("failed to get dump statistics: %w", err)
+			}
+
+			statsTree := tree.AddBranch("CRIU dump statistics")
+			statsTree.AddBranch(fmt.Sprintf("Freezing Time: %d us", dumpStats.GetFreezingTime()))
+			statsTree.AddBranch(fmt.Sprintf("Frozen Time: %d us", dumpStats.GetFrozenTime()))
+			statsTree.AddBranch(fmt.Sprintf("Memdump Time: %d us", dumpStats.GetMemdumpTime()))
+			statsTree.AddBranch(fmt.Sprintf("FMemwrite Time: %d us", dumpStats.GetMemwriteTime()))
+			statsTree.AddBranch(fmt.Sprintf("Pages Scanned: %d us", dumpStats.GetPagesScanned()))
+			statsTree.AddBranch(fmt.Sprintf("Pages Written: %d us", dumpStats.GetPagesWritten()))
+
+		}
+
+		if psTree {
+			// The image files reside in a subdirectory called "checkpoint"
+			c := crit.New("", "", filepath.Join(task.outputDir, "checkpoint"), false, false)
+			// Get process tree with CRIT
+			psTree, err := c.ExplorePs()
+			if err != nil {
+				return fmt.Errorf("failed to get process tree: %w", err)
+			}
+
+			pTree := tree.AddBranch("Processes")
```

Comment:
It would be better to name this `psTreeNode` instead. `pTree` and `psTree` would be too easy to mix up.